### PR TITLE
Implement api end points to list all notifications and mail-notifications (non-self, for admin)

### LINKED ISSUE #4237: Implement api end points to list all notifications and mail-notifications (non-self, for admin)
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

According to ,
- https://github.com/fossasia/open-event/blob/master/dev-handbook/resources-permissions.md#notifications---eg-v1notifications
- https://github.com/fossasia/open-event/blob/master/dev-handbook/resources-permissions.md#notifications---eg-v1notifications

### PR DESCRIPTION
**Pull request by shubham-padia**
fixes #4237

### CODE REVIEW

File: tests/hook_main.py [hunk 1]
```diff
@@ -1757,7 +1770,20 @@ def notification_delete(transaction):
 
 
 # ------------------------- Email Notifications -------------------------
-@hooks.before("Email Notifications > Email Notifications Collection Get > List All Email Notifications")
+@hooks.before("Email Notifications > Email Notifications Admin Collection > List All Email Notifications")
+def email_notification_get_admin_list(transaction):
+    """
+    GET email-notifications
```
Comment (enigmaeth) [inline tests/hook_main.py]: GET `/email-notifications` here :)

File: docs/api/api_blueprint.apib [hunk 2]
```diff
@@ -12149,94 +12206,72 @@ To turn Email Notifications ON/OFF related to the various events, session approv
...
-                    "self": "/v1/email-notifications/1"
                 }
-            },
+            ],
             "jsonapi": {
                 "version": "1.0"
             },
             "links": {
-                "self": "/v1/email-notifications/1"
+                "self": "/v1/users/2/email-notifications"
```
Comment (enigmaeth) [inline docs/api/api_blueprint.apib]: Same for self url here also please.

File: docs/api/api_blueprint.apib [hunk 3]
```diff
@@ -12009,6 +12009,62 @@ Get a list of Notifications.
...
+                    "links": {
+                        "self": "/v1/notifications/1"
+                    }
+                }
+            ],
+            "jsonapi": {
+                "version": "1.0"
+            },
+            "links": {
+                "self": "/v1/users/2/notifications"
```
Comment (enigmaeth) [inline docs/api/api_blueprint.apib]: Can you please check the self url here once again? I haven't checked it yet but I think it should be `/v1/notifications`.

Comment (shubham-padia) [inline docs/api/api_blueprint.apib]: @enigmaeth Resolved :)

File: docs/api/api_blueprint.apib (hunk 2 above)
Comment (shubham-padia) [inline docs/api/api_blueprint.apib]: @enigmaeth Resolved :)

File: tests/hook_main.py (hunk 1 above)
Comment (shubham-padia) [inline tests/hook_main.py]: @enigmaeth Resolved :)

**Pull request merged**